langchain4j-qianfan/src/main/java/dev/langchain4j/model/qianfan/QianfanChatModel.java

```
@@ -0,0 +1,137 @@
+                            Float topP,
+                            String modelName,
+                            String endpoint,
+                            String responseFormat,
+                            Float penaltyScore
+                             ) {
+        if (Utils.isNullOrBlank(apiKey)||Utils.isNullOrBlank(secretKey)) {
+            throw new IllegalArgumentException(" api key and secret key must be defined. It can be generated here: https://console.bce.baidu.com/qianfan/ais/console/applicationConsole/application");
+        }
+     
+        this.modelName=modelName;
+        this.endpoint=Utils.isNullOrBlank(endpoint)? QianfanModelEnum.getEndpoint(modelName):endpoint;
+
+        if (Utils.isNullOrBlank(this.endpoint) ) {
+            throw new IllegalArgumentException("Qianfan is no such model name. You can see model name here: https://cloud.baidu.com/doc/WENXINWORKSHOP/s/Nlks5zkzu");
+        }
+
+        this.baseUrl = getOrDefault(baseUrl,  "https://aip.baidubce.com");
+
+        this.client = QianfanClient.builder().baseUrl(this.baseUrl).apiKey(apiKey).secretKey(secretKey).logStreamingResponses(true).build();
+        this.temperature = getOrDefault(temperature, 0.7);
+        this.maxRetries = getOrDefault(maxRetries, 3);
+        this.topP = topP;
+        this.penaltyScore = penaltyScore;
+        this.responseFormat = responseFormat;
+    }
+
+
+    @Override
+    public Response<AiMessage> generate(List<ChatMessage> messages) {
+
+          return  generate(messages, null,null);
+    }
+
+    @Override
+    public Response<AiMessage> generate(List<ChatMessage> messages, List<ToolSpecification> toolSpecifications) {
+        return generate(messages, toolSpecifications,null);
+    }
+
+    @Override
+    public Response<AiMessage> generate(List<ChatMessage> messages, ToolSpecification toolSpecification) {
+        return generate(messages, singletonList(toolSpecification), toolSpecification);
```

```suggestion
        throw new RuntimeException("Not supported");
```
If there is no way to force the model to call specific tool/function, please throw an exception here instead of doing it later.